pkg/cloudprovider/aws/instancetype.go

```diff
@@ -83,6 +84,29 @@ func (i *InstanceType) NvidiaGPUs() *resource.Quantity {
 	return resources.Quantity(fmt.Sprint(count))
 }
 
+func NeedsGPUAmi(is []cloudprovider.InstanceType) bool {
+	for _, i := range is {
+		if !i.NvidiaGPUs().IsZero() || !i.AWSNeurons().IsZero() {
+			return true
+		}
+	}
+	return false
+}
+
+// NeedsDocker returns true if the instance type is unable to use
+// conatinerd directly
+func NeedsDocker(is []cloudprovider.InstanceType) bool {
```

Comment:
IMO these functions are better encapsulated by launchtemplate.go. Until now, this file was just a pure implementation of the cloudprovider.InstanceType interface. If you move them, I'd make the methods private, too.

Reply:
Sounds good.

Reply:
Done